### PR TITLE
Update calico postrelease tests [v3.27]

### PR DESCRIPTION
**Pull request by danudey**
Cherry-pick of #8552

### CODE REVIEW

File: hack/postrelease/tests/test_images.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+# a registry assigned.
+EXPECTED_IMAGES = [
+    "calico/node",
+    "calico/ctl",
+    "calico/apiserver",
+    "calico/typha",
+    "calico/cni",
+    "calico/kube-controllers",
+    "calico/upgrade",
+    "calico/windows",
```
Comment (lwr20) [inline hack/postrelease/tests/test_images.py:23]: This list appears to be missing the [`node-windows`](https://quay.io/repository/calico/node-windows) and [`cni-windows`](https://quay.io/repository/calico/cni-windows) images. 
```suggestion
    "calico/windows",
    "calico/windows-upgrade",
    "calico/node-windows",
    "calico/cni-windows",
```

`windows-upgrade` and `windows` images might be obsolete now that we use host-process-container install.  @coutinhop would know for sure.

File: hack/postrelease/Makefile [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	@touch .docker_container.stamp
+
+## Execution
+
+run_pytests_docker: create_docker_container require-vars-QUAY_TOKEN
+	$(info +++ Running postrelease tests)
+	@docker run -it --rm                     \
+		-v .:/code                           \
+		-v /run/docker.sock:/run/docker.sock \
+		-e QUAY_TOKEN                        \
```
Comment (lwr20) [inline hack/postrelease/Makefile:49]: Why is a quay token needed?  These images need to be publicly available, so I think we should be testing without a quay-token.

File: hack/postrelease/tests/test_images.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        tag_suffixed = f"{variables.RELEASE_VERSION}-{tag_suffix}"
+        ALL_IMAGES.append((image, tag_suffixed))
+
+print("Found {} separate image tags to test".format(len(ALL_IMAGES)))
+
+
+def request_quay_image(image_name, image_version):
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": "Bearer {}".format(variables.QUAY_TOKEN),
```
Comment (lwr20) [inline hack/postrelease/tests/test_images.py:72]: We should be testing this without a quay token.

File: hack/postrelease/tests/variables.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+
+# Get flannel version.
+FLANNEL_VERSION = get_var_or_error("FLANNEL_VERSION")
+
+# Get expected operator version.
+OPERATOR_VERSION = get_var_or_error("OPERATOR_VERSION")
+
+# Quay.io API token
+QUAY_TOKEN = get_var_or_error("QUAY_TOKEN")
```
Comment (lwr20) [inline hack/postrelease/tests/variables.py:35]: We should be testing these without a quay token.

File: hack/postrelease/tests/test_images.py (hunk 1 above)
Comment (coutinhop) [reply hack/postrelease/tests/test_images.py:23]: Yes, that's correct, `calico/windows` and `calico/windows-upgrade` are obsolete, so we should only have `calico/node-windows` and `calico/cni-windows`

Comment (coutinhop) [reply hack/postrelease/tests/test_images.py:23]:
```suggestion
    "calico/cni-windows",
    "calico/node-windows",
```

**Review by matthewdupre — APPROVED** [review]

**Pull request merged**